docs/rules/semi-spacing.md

```diff
@@ -0,0 +1,87 @@
+# Enforce spacing before and after semicolons (semi-spacing)
+
+JavaScript allows you to place unnecessary spaces before or after a semicolon.
+
+Disallowing or enforcing space around a semicolon can improve the readability of your program.
+
+```js
+var a = "b" ;
+
+var c = "d";var e = "f";
+```
+
+## Rule Details
+
+This rule aims to enforce spacing around a semicolon. This rule prevents the use of spaces before a semicolon in expressions.
+
+### Options
+
+The rule takes one option, an object, which has two keys `before` and `after` having boolean values `true` or `false`.
+If `before` is `true`, space is enforced before semicolons and if it's `false`, space is disallowed before semicolons.
+If `after` is `true`, space is enforced after semicolons and if it's `false`, space is disallowed after semicolons.
```

Comment:
We should mention here that `after` is only applied when the semicolon is not the last token in its line.